### PR TITLE
[chore] Import style guide from examples submodule

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Symlink the style guides from the examples submodule into root as needed
  - Can't indicate `.editorconfig` path for linter, so need to make file present in root.
- Update Make steps, CI as needed to account for style guide imports

# Testing

- Make steps work as expected locally

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -52,8 +52,15 @@ install-tools:
 install-release-tools:
 	bash scripts/unix/install_osslsigncode.sh
 
+## install-style - Import style guide (Unix only)
+install-style: | update-examples-submodule
+	sh examples/symlink_directory_files.sh examples/style_guides/csharp .
```
Comment (Justintime50) [inline Makefile:57]: praise: cool, we can reuse the script here too

File: Makefile [hunk 2]
```diff
@@ -52,8 +52,15 @@ install-tools:
...
+
 ## install - Install requirements
-install: | install-tools
+install: | install-tools update-examples-submodule
+
+## update-examples-submodule - Update the examples submodule
+update-examples-submodule:
```
Comment (Justintime50) [inline Makefile:63]: praise: smart abstraction

File: .gitignore [hunk 3]
```diff
@@ -176,3 +176,7 @@ $RECYCLE.BIN/
 
 # Dotnet tools
 .config/dotnet-tools.json
+
+# Downloaded style guides
+/layout_rules.xml
+/.editorconfig
```
Comment (Justintime50) [inline .gitignore:182]: nit: newline please

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -18,7 +18,7 @@ jobs:
           dotnet-version: 7.x.x
 
       - name: Set up dotnet tools
-        run: make install-tools
+        run: make install install-style
```
Comment (Justintime50) [inline .github/workflows/ci.yml:21]: thought: prefer `install-styleguide`?

**Review by Justintime50 — APPROVED** [review]

**Review by Justintime50 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]

**Pull request merged**